iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -436,62 +326,23 @@ public static Anonymizer write(Network network, ExportOptions options, DataSourc
-                if (options.isAnonymized()) {
-                    try (BufferedWriter writer2 = new BufferedWriter(new OutputStreamWriter(dataSource.newOutputStream("_mapping", "csv", false), StandardCharsets.UTF_8))) {
-                        anonymizer.write(writer2);
-                    }
-                }
-                return anonymizer;
-            }
-
-            NetworkXmlWriterContext context = writeBaseNetwork(network, bosb, options);
-            writeEndElement(context.getWriter());
-
-            // write extensions
-            if (!options.withNoExtension() && !getNetworkExtensions(network).isEmpty()) {
-
-                if (options.getMode() == IidmImportExportMode.EXTENSIONS_IN_ONE_SEPARATED_FILE) {
-                    try (OutputStream ose = dataSource.newOutputStream("-ext", dataSourceExt, false);
-                         BufferedOutputStream bose = new BufferedOutputStream(ose)) {
-
-                        final XMLStreamWriter extensionsWriter = initializeWriter(network, bose, options);
-                        context.setExtensionsWriter(extensionsWriter);
-                        writeExtensions(network, context, options);
-                        writeEndElement(extensionsWriter);
-                    }
-                } else if (options.getMode() == IidmImportExportMode.ONE_SEPARATED_FILE_PER_EXTENSION_TYPE) {
-                    writeExtensionsInMultipleFile(network, context, dataSource, options, dataSourceExt);
-                }
-            }
-
+            Anonymizer anonymizer = write(network, options, bosb);
             if (options.isAnonymized()) {
                 try (BufferedWriter writer2 = new BufferedWriter(new OutputStreamWriter(dataSource.newOutputStream("_mapping", "csv", false), StandardCharsets.UTF_8))) {
-                    context.getAnonymizer().write(writer2);
+                    anonymizer.write(writer2);
                 }
             }
-            return context.getAnonymizer();
-        } catch (XMLStreamException e) {
-            throw new UncheckedXmlStreamException(e);
+            return anonymizer;
         }
     }
 
     public static Anonymizer writeAndValidate(Network n, Path xmlFile) {
-        try {
-            return writeAndValidate(n, new ExportOptions(), xmlFile);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
+        return writeAndValidate(n, new ExportOptions(), xmlFile);
     }
 
-    public static Anonymizer writeAndValidate(Network n, ExportOptions options, Path xmlFile) throws IOException {
+    public static Anonymizer writeAndValidate(Network n, ExportOptions options, Path xmlFile) {
         Anonymizer anonymizer = write(n, options, xmlFile);
-        if (options.getMode() == IidmImportExportMode.UNIQUE_FILE) {
-            validateWithExtensions(xmlFile);
-            return anonymizer;
-        }
-        validate(xmlFile, options.getMode());
+        validateWithExtensions(xmlFile);
```

Comment:
Is there a validateWithoutExtension?

Reply:
There was but as it is not used anymore, I will rename this one as you suggested